### ISSUE TITLE
many of the Copilot Agent responses are not accessible to a screen reader and the reply "inline" is a very poor keyboarding experience 

### ISSUE THREAD

**Issue by Electro-Jam** [microsoft/vscode#252090]
Type: <b>Bug</b>

I use a screen reader to hear the responses as I go back and forth with Copilot.  I am trying out the use of the Agent mode and ChatGPT 4.1.   In this case I am giving it a task to reivew what needs to be checked in, create a commit, and let me review it before it goes in.  

Many of the replies are not being read back to the screen reader (in this case i am using NVDA). 

Once there is a reply getting to the options such as "continue" or "cancel" is a LOT of keyboard work.  I would like to simply be able to reply and say continue or cancle.

I will add a few smapshots so you can see the issue.

VS Code version: Code 1.101.1 (18e3a1ec544e6907be1e944a94c496e302073435, 2025-06-18T13:35:12.605Z)
OS version: Windows_NT x64 10.0.26100
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|AMD Ryzen 7 Microsoft Surface (R) Edition (16 x 1996)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: enabled_on<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_graphite: disabled_off<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled<br>webnn: disabled_off|
|Load (avg)|undefined|
|Memory (System)|15.45GB (6.54GB free)|
|Process Argv|--crash-reporter-id 114a894a-a9e4-4018-8946-bd9bb51920b9|
|Screen Reader|yes|
|VM|0%|
</details><details><summary>Extensions (54)</summary>

Extension|Author (truncated)|Version
---|---|---
better-comments|aar|3.0.2
visubezier|chr|1.6.1
doxdocgen|csc|1.4.0
gtest-adapter|Dav|1.8.3
EditorConfig|Edi|0.17.4
prettier-vscode|esb|11.0.0
copilot|Git|1.336.0
copilot-chat|Git|0.28.1
vscode-codeowners|jas|1.1.1
better-cpp-syntax|jef|1.27.1
vscode-peacock|joh|4.2.2
cmake-language-support-vscode|jos|0.0.9
node-module-intellisense|lei|1.5.0
cortex-debug|mar|1.12.1
debug-tracker-vscode|mcu|0.0.15
memory-view|mcu|0.0.25
peripheral-viewer|mcu|1.4.6
rtos-views|mcu|0.0.7
vscode-language-babel|mgm|0.0.40
dotenv|mik|1.0.1
csharp|ms-|2.80.16
vscode-dotnet-runtime|ms-|2.3.5
debugpy|ms-|2025.8.0
isort|ms-|2025.0.0
python|ms-|2025.8.0
vscode-pylance|ms-|2025.6.1
jupyter|ms-|2025.5.0
jupyter-keymap|ms-|1.1.2
jupyter-renderers|ms-|1.1.0
vscode-jupyter-cell-tags|ms-|0.1.9
vscode-jupyter-slideshow|ms-|0.1.6
remote-containers|ms-|0.417.0
remote-ssh|ms-|0.120.0
remote-ssh-edit|ms-|0.87.0
remote-wsl|ms-|0.99.0
cmake-tools|ms-|1.20.53
cpptools|ms-|1.25.3
cpptools-extension-pack|ms-|1.3.1
makefile-tools|ms-|0.12.17
remote-explorer|ms-|0.5.0
vscode-serial-monitor|ms-|0.13.1
nvs-extention-pack|net|0.0.4
nvs-vue-snippets|net|1.0.3
colorful-comments|Par|1.0.0
markdown-preview-enhanced|shd|0.8.18
cmake|twx|0.0.17
vscode-svg-previewer|vit|0.7.0
volar|Vue|2.2.10
vscode-todo-highlight|way|1.0.5
twig|wha|1.0.2
vscode-import-cost|wix|3.3.0
clang-format|xav|1.9.0
zeplin|Zep|2.2.0
linkerscript|Zix|1.0.4

(1 theme extensions excluded)

</details><details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
vspor879:30202332
vspor708:30202333
vspor363:30204092
vscod805cf:30301675
binariesv615:30325510
c4g48928:30535728
azure-dev_surveyone:30548225
2i9eh265:30646982
962ge761:30959799
h48ei257:31000450
pythontbext0:30879054
cppperfnew:31000557
dwnewjupyter:31046869
pythonrstrctxt:31112756
nativeloc2:31192216
5fd0e150:31155592
dwcopilot:31170013
6074i472:31201624
dwoutputs:31242946
customenabled:31248079
9064b325:31222308
copilot_t_ci:31222730
e5gg6876:31282496
pythoneinst12:31285622
bgtreat:31268568
4gafe986:31271826
c7cif404:31314491
e77a2192:31326280
pythonpulldiagcf:31325931
996jf627:31283433
pythonrdcb7:31303018
usemplatestapi:31297334
0aa6g176:31307128
7bj51361:31289155
747dc170:31275177
6518g693:31325878
aj953862:31281341
generatesymbolt:31295002
convertfstringf:31295003
0g0a1943:31332226
d7aab740:31332224
usemarketplace:31331644
agentclaude:31331946
id45c886:31333049
no_f_c:31333023

```

</details>

**Comment by Electro-Jam** [comment]
Here is what I have in my git repo:
A new docs directory 
inside a few new CLAUDE.md file 

When asking to review the files that need to be checked in I started with:
can you please reivew what files need to be checked in and le tme know what they are please.  Do not do anything else at this time.

BUG: The reply does not read to a screen reader user:

![Image](https://github.com/user-attachments/assets/c06d25d7-1734-4655-b51f-8824d20c2c73)

When I go on to ask: please create the add and commit with the commit message and confirm with me before the commit.

The following response is not read out by NVDA:

![Image](https://github.com/user-attachments/assets/f55241fe-6abf-4eee-9d3b-050b1fba3fbe)

Getting to the buttons for "continue" / "cancel" is super complicated from a keyboard point of view.   For this case I simply typed "continue" and a new message came up that was not read to the screen reader:

![Image](https://github.com/user-attachments/assets/73cf0894-29f0-436e-8016-3a2114237c4b)

Overall this workflow is not accessible and needs work as soon as possible please.

**Comment by meganrogge** [comment]
Thanks for the issue and the feedback. `workbench.action.chat.acceptTool` with `ctrlCmd+enter` from within the chat input will continue on. I'll add that to the accessibility help dialog using this issue to track that. 

You can also set `chat.tools.autoApprove` if you want auto approval, which I'll also add to the accessibility help dialog.

 For me, the response is alerted when a confirmation message appears it says `Run command in terminal... terminal command` for example. It can also be accessed via `ctrlCmd+upArrow`.

**Comment by roblourens** [comment]
Seems like we now mention `workbench.action.chat.acceptTool` twice:

```
To take the action, use the accept tool command (Command+Enter).
...
To accept a tool action, use the Accept Tool Confirmation command (Command+Enter).
```
